### PR TITLE
Finalize Exonum-derive macros [ECR-3800]

### PR DESCRIPTION
**Pull request by alekseysidorov**
This PR contains two main changes:

- Extract `ServiceDispatcher` derivation from the `ServiceFactory` into the separate macro.
- Rename `exonum` attribute in the `IntoExecutionError` derive macro.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -675,7 +675,7 @@ The project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html)
     ```
 
     where `MyQueryType` type implements `Deserialize` trait and `MyResponse`
-    implements `Serialize` trait.
+)]#[service_dispatcher( implements `Serialize` trait.
```
Comment (popzxc) [inline CHANGELOG.md]: ?

File: components/derive/src/lib.rs [hunk 2]
```diff
@@ -74,54 +75,74 @@ pub fn object_hash(input: TokenStream) -> TokenStream {
...
+/// # Examples TODO Move to ServiceFactory documentation [ECR-3275]
 ///
 /// Typical usage.
 /// ```ignore
 /// #[derive(ServiceFactory)]
-/// #[exonum(
+/// #[service_factory(
 ///     proto_sources = "crate::proto",
-///     implements("MyServiceInterface")
+/// )]#[service_dispatcher( implements("MyServiceInterface")
```
Comment (popzxc) [inline components/derive/src/lib.rs]: Missing newline?

File: components/derive/src/lib.rs [hunk 3]
```diff
@@ -139,10 +139,10 @@ pub fn service_dispatcher(input: TokenStream) -> TokenStream {
 ///
 /// Typical usage.
 /// ```ignore
+/// #[service_dispatcher(implements("MyServiceInterface"))]
```
Comment (popzxc) [inline components/derive/src/lib.rs]: IMO having `derive` between two proc-macros is a bit weird.

File: components/derive/src/lib.rs [hunk 4]
```diff
@@ -159,15 +178,15 @@ pub fn object_hash(input: TokenStream) -> TokenStream {
...
 /// ```
-#[proc_macro_derive(ServiceFactory, attributes(exonum))]
-pub fn generate_service_factory(input: TokenStream) -> TokenStream {
-    service_factory::implement_service_factory(input)
+#[proc_macro_derive(ServiceFactory, attributes(service_factory))]
+pub fn service_factory(input: TokenStream) -> TokenStream {
+    service_factory::impl_service_factory(input)
 }
 
 /// Mark trait as an Exonum service interface.
```
[the remark this answers is not in the document]
Comment (alekseysidorov) [reply components/derive/src/lib.rs]: In my opinion this should be part of the ECR-3822 task.

File: components/derive/src/lib.rs [hunk 5]
```diff
@@ -71,7 +72,27 @@ pub fn binary_value(input: TokenStream) -> TokenStream {
...
+
+/// Derive `ServiceDispatcher` trait.
+///
+/// # Attributes:
+///
+/// ## Required
+///
+/// * `#[service_dispatcher(implements(""path_1", "path_2""))]`
+///
+/// Path list to the interfaces which have been implemented by the service.
```
[the remark this answers is not in the document]
Comment (alekseysidorov) [reply components/derive/src/lib.rs]: Also I am going to fix this comments as part of the  ECR-3822 task.

File: components/derive/src/lib.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (popzxc) [reply components/derive/src/lib.rs]: Sure, I don't mind.

**Pull request merged**